compiler/rustc_middle/src/ty/closure.rs

```diff
@@ -96,15 +96,7 @@ impl<'tcx> ClosureKind {
     /// Returns `true` if a type that impls this closure kind
     /// must also implement `other`.
     pub fn extends(self, other: ty::ClosureKind) -> bool {
-        matches!(
-            (self, other),
-            (ClosureKind::Fn, ClosureKind::Fn)
-                | (ClosureKind::Fn, ClosureKind::FnMut)
-                | (ClosureKind::Fn, ClosureKind::FnOnce)
-                | (ClosureKind::FnMut, ClosureKind::FnMut)
-                | (ClosureKind::FnMut, ClosureKind::FnOnce)
-                | (ClosureKind::FnOnce, ClosureKind::FnOnce)
-        )
+        self <= other
```

Comment:
can you add a comment on `ClosureKind` that the `Ord` implementation is load-bearing? this seems easy to break (though I guess it doesn't matter, since I doubt `core` would even bootstrap if this got messed up).

Reply:
There is already a comment:
https://github.com/rust-lang/rust/blob/3d407cc1131b3e3733f1f7a2fa0783fec9d70dde/compiler/rustc_middle/src/ty/closure.rs#L84-L86